src/test/functional/tests/buzzPage.spec.ts

```diff
@@ -20,21 +30,68 @@ test.describe("Buzz Page", () => {
 
   test.afterEach(async ({ page }) => {
     buzzPage = new BuzzPage(page);
-    await buzzPage.deleteTheNewestPost()
+    await buzzPage.deleteTheNewestPost(false)
   });
 
   test("Post should be shared", async ({ page }) => {
+    const randomTitle = generateRandomString()
     await buzzPage.navigateToSubPage(buzzPage.buzzPageButton);
-    await buzzPage.sharePost(filePath);
+    await buzzPage.sharePost(filePath, randomTitle);
     await page.reload();
-    await expect(buzzPage.photoImg.first()).toBeAttached();
+    await expect(buzzPage.bodyTextPost.first()).toHaveText(randomTitle);
   });
 
   test("Post should be edited", async ({ page }) => {
+    const randomTitle = generateRandomString()
     await buzzPage.navigateToSubPage(buzzPage.buzzPageButton);
-    await buzzPage.sharePost(filePath);
+    await buzzPage.sharePost(filePath, randomTitle);
     await page.reload();
     await buzzPage.editTheNewestPost(expectedPostTextAfterEdition)
-    await expect(page.getByText(expectedPostTextAfterEdition)).toBeVisible();
+    await expect(buzzPage.bodyTextPost.first()).toHaveText(expectedPostTextAfterEdition);
+  });
+
+  test("Video should be shared", async ({ page }) => {
+    await buzzPage.navigateToSubPage(buzzPage.buzzPageButton);
+    await buzzPage.shareVideo(videoTitle, videoUrl);
+    await expect(buzzPage.bodyTextPost.first()).toHaveText(videoTitle);
+  });
+});
+
+test.describe("Most liked and most commented post", () => {
+  let loginPage: LoginPage;
+  let buzzPage: BuzzPage;
+
+  test.beforeEach(async ({ page }) => {
+    loginPage = new LoginPage(page);
+    buzzPage = new BuzzPage(page);
+    await loginPage.initialize();
+    await loginPage.navigateToMainPage();
+    await loginPage.loginUser(adminUserTestData.userName, adminUserTestData.password);
+    await buzzPage.navigateToSubPage(buzzPage.buzzPageButton);
+    await buzzPage.shareVideo(videoTitle, videoUrl);
+    await buzzPage.shareVideo(mostLiked, videoUrl2);
+    await buzzPage.shareVideo(mostCommented, videoUrl3);
+    
+  });
+
+  test.afterEach(async ({ page }) => {
+    buzzPage = new BuzzPage(page);
+    await buzzPage.deleteTheNewestPost(true)
+    await buzzPage.deleteTheNewestPost(true)
+    await buzzPage.deleteTheNewestPost(true)
```

Comment:
what If test failed and there will be more than 3 newest posts?

Reply:
there is always risk do you have better idea? cleaning of tests' data should be performed by cleaning of database or at least by api request that are faster and more stable, but we have restriction here.

Reply:
deleting in the loop?

Reply:
how can i get the length do you have any idea?